src/classes/ALLO_PaymentSync_TEST.cls

```diff
@@ -832,6 +832,78 @@ private with sharing  class ALLO_PaymentSync_TEST {
         Test.stopTest();
     }
 
+    /*******************************************************************************************************
+    * @description Test when an opportunity is fully scheduled with one paid payment and one scheduled payment,
+    * updating the allocations should not change the paid payment allocation. However, it should update the
+    * schedule payment's allocations.
+    ********************************************************************************************************/
+    @isTest
+    private static void testSyncOpportunityToPaymentWithFullyScheduledPartiallyPaid() {
+        setupForTests();
+        makeGAUs(1);
+        opportunity = [SELECT Id, Name, Amount FROM Opportunity LIMIT 1];
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        alloForInsert.add(new Allocation__c(Opportunity__c=opportunity.Id, Percent__c = 100, General_Accounting_Unit__c = gaus[0].Id));
+        insert alloForInsert;
+
+        makePayments(1000, 2);
+
+        npe01__OppPayment__c paidPayment = new npe01__OppPayment__c(Id = payments[0].Id, npe01__Paid__c = true);
+        update paidPayment;
+
+        Allocation__c oppAllocation = [select Id,
+                Opportunity__c,
+                Payment__c,
+                General_Accounting_Unit__c,
+                General_Accounting_Unit__r.Name,
+                Amount__c,
+                Percent__c
+        from Allocation__c
+        where Opportunity__c = :opportunity.Id
+        AND General_Accounting_Unit__c =: gaus[0].Id
+        order by Amount__c DESC];
+
+        oppAllocation.Percent__c = 50;
+
+        Test.startTest();
+        update oppAllocation;
+
+        List<Allocation__c> paidPaymentAllocationList = new List<Allocation__c>([select Id,
```

Comment:
I personally prefer saying List, Set, Map, etc. as you did, but that is not our standard.  Our standard is just to pluralize collections e.g. paidPaymentAllocations, or if it were a map paidPaymentAllocationsById

Reply:
Updated